### DISCUSSION TITLE
Packmol-memgen ensambles with sirah flags -> ERROR: atom selection with index greater than number of atoms in structure  3

### DISCUSSION THREAD

**Discussion by b-lillo** [SIRAHFF/documentation#86] [General]
Hello, nice to greet you,
I very recently started using coarse-grained simulations to extend the exploration of conformational change in a transporter. However after using the following code:
**[packmol-memgen -p best_model_protonate.pdb --sirah --verbose]**
I am encountering an error in packmol.log:
**“ERROR: atom selection with index greater than number of atoms in structure 3”.**
To find more information I analyze the file **packmol.inp:**
structure siDOPC.pdb
  nloop 20
  number 151
  inside box -51.18 -51.18 -23.0 51.18 51.18 0.0
  atoms 8
    below plane 0. 0. 0. 1. -16.0
  end atoms
  atoms 6 17
    over plane 0. 0. 0. 1. -4.0
  end atoms
end structure

I understand that there is a restriction for the correct positioning of the bilayers, but according to si.DOPC.pdb (file used to create the assembly) it has only two atoms (xOL and xPC). Can you help me?
Python 3.12.2, ambertools 23.6, packmol-memgen 2024.2.9 all installed in conda environment.  

Translated with DeepL.com (free version)

**Comment by luciannahss** [comment]
Hi b-lillo,

The error appears to be a result of the input you are providing to Packmol-memgen. You are giving index numbers (atoms 8 / atoms 6 17) that the program is not finding in the PDB file.

Regrettably, we, from the SIRAH Team, lack experience with Packmol-memgen. It would be more beneficial to you to go to the Amber or Packmol-memgen's official forums to get more information.

In our tutorials (AMBER's tutorials 6 and 7), we suggest that you utilize PACKMOL (https://m3g.github.io/packmol/) to construct the system with the reference building-block structures that are available in the sirah.ff/PDB/ folder. The mapping scheme employed by CGConv is consistent with the build blocks in the folder.  

You can also use the Charmm-gui server to create your system. But bear in mind that to convert from AA to CG you need to use the correct mapping scheme for the lipids you are using (see tutorials 6 and 7).

I hope this was helpful, but if not, feel free to ask us again.
Lucianna and the SIRAH Team.
